api/src/main/java/bio/terra/tanagra/query/CellValue.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.tanagra.query;
+
+import bio.terra.tanagra.underlay.Literal;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+/**
+ * An interface for the value of a cell within a row within a result table.
+ *
+ * <p>This interface allows us to read data from different databases in a simple but uniform way.
+ * Different database types should implement this for returning values.
+ */
+public interface CellValue {
+  /** Enum for the SQL data types supported by Tanagra. */
+  enum SQLDataType {
```

Comment:
I think it's better to decouple Tanagra's data types from the underlying data storage's data types, so that we can e.g. decide to treat timestamps as strings or dates in Tanagra's API. In the future, we may also want to provide different conversions for different backend data sources.

Reply:
Shouldn't this be in the BigQuery implementation of this interface then instead?